src/steadystateproblem.cpp

```diff
@@ -363,37 +411,40 @@ bool SteadystateProblem::getSensitivityFlag(const Model *model,
 realtype SteadystateProblem::getWrmsNorm(const AmiVector &x,
                                          const AmiVector &xdot,
                                          realtype atol,
-                                         realtype rtol) {
+                                         realtype rtol,
+                                         AmiVector &ewt) {
     N_VAbs(x.getNVector(), ewt.getNVector());
     N_VScale(rtol, ewt.getNVector(), ewt.getNVector());
     N_VAddConst(ewt.getNVector(), atol, ewt.getNVector());
     N_VInv(ewt.getNVector(), ewt.getNVector());
     return N_VWrmsNorm(xdot.getNVector(), ewt.getNVector());
 }
 
-bool SteadystateProblem::checkConvergence(const Solver *solver, Model *model) {
+bool SteadystateProblem::checkConvergence(const Solver *solver,
+                                          Model *model,
+                                          SensitivityMethod checkSensitivities) {
     /* get RHS and compute weighted error norm */
-    model->fxdot(t, x, dx, xdot);
-    wrms = getWrmsNorm(x, xdot, solver->getAbsoluteToleranceSteadyState(),
-                       solver->getRelativeToleranceSteadyState());
+    if (checkSensitivities == SensitivityMethod::adjoint) {
+        getQBfromQ(model, xQ, xQB);
+        getQBfromQ(model, xB, xQBdot);
```

Comment:
Originally, I wanted to avoid computing these two thing in every step. I thought integrating xB to xQ and then multiplying with dxdotdp to get xQB is enough, because I hoped that xQBdot and xQdot (=xB) always become small together. However, xB can remain strictly positive, while xQBdot goes to zero. So we need to really check the whole xQBdot thing here...